### PR TITLE
Update proto constructor to handle nested non-built in types

### PR DESCRIPTION
**Pull request by justinchou-google**
Resolves bug where nested array non-built in types are not correctly being constructed.
b/242735734

### CODE REVIEW

File: src/proto/api_messages.js [hunk 1]
```diff
@@ -2279,7 +2287,9 @@ class SurveyDataTransferRequest {
     const base = includesLabel ? 1 : 0;
 
     /** @private {!Array<!SurveyQuestion>} */
-    this.surveyQuestions_ = data[base] || [];
+    this.surveyQuestions_ = (data[base] || []).map(
+      (item) => new SurveyQuestion(item, includesLabel)
+    );
```
Comment (justinchou-google) [inline src/proto/api_messages.js:2292]: change #1

File: src/proto/api_messages.js [hunk 2]
```diff
@@ -2394,7 +2406,9 @@ class SurveyQuestion {
     this.questionCategory_ = data[2 + base] == null ? null : data[2 + base];
 
     /** @private {!Array<!SurveyAnswer>} */
-    this.surveyAnswers_ = data[3 + base] || [];
+    this.surveyAnswers_ = (data[3 + base] || []).map(
+      (item) => new SurveyAnswer(item, includesLabel)
+    );
```
Comment (justinchou-google) [inline src/proto/api_messages.js:2411]: change #2

**Comment by ChrisAntaki** [comment]
Just curious, why are there so many formatting changes?

File: src/proto/api_messages-test.js [hunk 3]
```diff
@@ -1459,205 +1459,6 @@ describe('SurveyQuestion', () => {
   });
 });
 
-describe('SurveyAnswer', () => {
```
Comment (justinchou-google) [inline src/proto/api_messages-test.js:1462]: duplicate testcase

**Comment by justinchou-google** [comment]
> Just curious, why are there so many formatting changes?

Updated, seems like there was some superfluous auto formatting being done when I copied the generated code in.

File: src/proto/api_messages-test.js (hunk 3 above)
Comment (ChrisAntaki) [reply src/proto/api_messages-test.js:1462]: Makes sense 👍

**Review by ChrisAntaki — APPROVED** [review]

**Pull request merged**